src/operator/aggregate/fold.rs

```diff
@@ -0,0 +1,86 @@
+use crate::{algebra::MonoidValue, operator::aggregate::Aggregator, trace::Cursor, Timestamp};
+use std::convert::identity;
+
+/// An [aggregator](`crate::operator::Aggregator`) that can be expressed
+/// as a fold of the input Z-set.
+///
+/// The `Fold` aggregator iterates over `(key, weight)` pairs with
+/// non-zero weights updating the value of an accumulator with a
+/// user-provided `step` function.  The final aggregate is produced by
+/// applying a user-provided `output` function to the final value of
+/// the accumulator.
+///
+/// # Type arguments
+///
+/// * `A` - accumulator
+/// * `SF` - step function
+/// * `OF` - output function
+#[derive(Clone)]
+pub struct Fold<A, SF, OF> {
+    init: A,
+    step: SF,
+    output: OF,
+}
+
+impl<A, SF> Fold<A, SF, fn(A) -> A> {
+    /// Create a `Fold` aggregator with initial accumulator value `init`,
+    /// step function `step`, and identity output function.
+    ///
+    /// This constructor caters for the common case when the final value of
+    /// the accumulator is the desired aggregate, i.e., no output processing
+    /// is required.
+    pub fn new(init: A, step: SF) -> Self {
+        Self {
+            init,
+            step,
+            output: identity,
+        }
+    }
+}
+
+impl<A, SF, OF> Fold<A, SF, OF> {
```

Comment:
How about a parallel composition of fold which makes a fold on tuples out of n folds point wise? May need to be a macro.

Reply:
Yep, that make sense. I'll create an issue (need to work on something else now).